### PR TITLE
add TiFlash spill doc

### PR DESCRIPTION
**Pull request by shichun-0415**
### First-time contributors' checklist 

- [ ] I've signed [**Contributor License Agreement**](https://cla-assistant.io/pingcap/docs) that's required for repo owners to accept my contribution.

### What is changed, added or deleted? (Required)
Add three tiflash configurations.

### Which TiDB version(s) do your changes apply to? (Required)

**Tips for choosing the affected version(s):**

By default, **CHOOSE MASTER ONLY** so your changes will be applied to the next TiDB major or minor releases. If your PR involves a product feature behavior change or a compatibility change, **CHOOSE THE AFFECTED RELEASE BRANCH(ES) AND MASTER**.

For details, see [tips for choosing the affected versions](https://github.com/pingcap/docs/blob/master/CONTRIBUTING.md#guideline-for-choosing-the-affected-versions).

- [x] master (the latest development version)
- [x] v7.0 (TiDB 7.0 versions)
- [ ] v6.6 (TiDB 6.6 versions)
- [ ] v6.5 (TiDB 6.5 versions)
- [ ] v6.4 (TiDB 6.4 versions)
- [ ] v6.1 (TiDB 6.1 versions)
- [ ] v5.4 (TiDB 5.4 versions)
- [ ] v5.3 (TiDB 5.3 versions)
- [ ] v5.2 (TiDB 5.2 versions)
- [ ] v5.1 (TiDB 5.1 versions)
- [ ] v5.0 (TiDB 5.0 versions)

### What is the related PR or file link(s)?

- This PR is translated from: https://github.com/pingcap/docs-cn/pull/13342
- Other reference link(s):

### Do your changes match any of the following descriptions?

- [ ] Delete files
- [ ] Change aliases
- [ ] Need modification after applied to another branch 
- [ ] Might cause conflicts after applied to another branch

### CODE REVIEW

File: tiflash/tiflash-configuration.md [hunk 1]
```diff
@@ -212,6 +212,14 @@ delta_index_cache_size = 0
     ## New in v6.2.0. This item specifies the minimum ratio of valid data in a PageStorage data file. When the ratio of valid data in a PageStorage data file is less than the value of this configuration, GC is triggered to compact data in the file. The default value is 0.5.
     dt_page_gc_threshold = 0.5
 
+    ## New in v7.0.0. This item specifies the maximum memory available for the HashAggregation operator with group by key before a spill is triggered. When the memory usage exceeds the threshold, HashAggregation reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited.
```
Comment (dveeden) [inline tiflash/tiflash-configuration.md:215]:
```suggestion
    ## New in v7.0.0. This item specifies the maximum memory available for the HashAggregation operator with group by key before a disk spill is triggered. When the memory usage exceeds the threshold, HashAggregation reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited and spill to disk is never used for HashAggregation.
```

File: tiflash/tiflash-configuration.md [hunk 2]
```diff
@@ -212,6 +212,14 @@ delta_index_cache_size = 0
...
+    max_bytes_before_external_group_by = 0
+
+    ## New in v7.0.0. This item specifies the maximum memory available for the sort or topN operator before a spill is triggered. When the memory usage exceeds the threshold, the sort or topN operator reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited.
```
Comment (dveeden) [inline tiflash/tiflash-configuration.md:218]:
```suggestion
    ## New in v7.0.0. This item specifies the maximum memory available for the sort or topN operator before a disk spill is triggered. When the memory usage exceeds the threshold, the sort or topN operator reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited and spill to disk is never used for topN.
```

File: tiflash/tiflash-configuration.md [hunk 3]
```diff
@@ -212,6 +212,14 @@ delta_index_cache_size = 0
...
+    max_bytes_before_external_sort = 0
+
+    ## New in v7.0.0. This item specifies the maximum memory available for the HashJoin operator with Equi Join before a spill is triggered. When the memory usage exceeds the threshold, HashJoin reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited.
```
Comment (dveeden) [inline tiflash/tiflash-configuration.md:221]:
```suggestion
    ## New in v7.0.0. This item specifies the maximum memory available for the HashJoin operator with EquiJoin before a disk spill is triggered. When the memory usage exceeds the threshold, HashJoin reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited and spill to disk is never used for HashJoin with EquiJoin.
```

File: tiflash/tiflash-configuration.md (hunk 1 above)
Comment (shichun-0415) [reply tiflash/tiflash-configuration.md:215]: @windtalker PTAL

Comment (windtalker) [reply tiflash/tiflash-configuration.md:215]: LGTM

File: tiflash/tiflash-configuration.md (hunk 2 above)
Comment (windtalker) [reply tiflash/tiflash-configuration.md:218]: Should be "spill to disk is never used for sort or TopN"

File: tiflash/tiflash-configuration.md (hunk 3 above)
Comment (windtalker) [reply tiflash/tiflash-configuration.md:221]: LGTM

File: tiflash/tiflash-configuration.md [hunk 4]
```diff
@@ -212,6 +212,14 @@ delta_index_cache_size = 0
...
+    max_bytes_before_external_sort = 0
```
Comment (shichun-0415) [inline tiflash/tiflash-configuration.md:222]:
```suggestion
    max_bytes_before_external_join = 0
```

Comment (windtalker) [inline tiflash/tiflash-configuration.md:222]: Should be `max_bytes_before_external_join`

File: tiflash/tiflash-configuration.md [hunk 5]
```diff
@@ -212,6 +212,14 @@ delta_index_cache_size = 0
     ## New in v6.2.0. This item specifies the minimum ratio of valid data in a PageStorage data file. When the ratio of valid data in a PageStorage data file is less than the value of this configuration, GC is triggered to compact data in the file. The default value is 0.5.
     dt_page_gc_threshold = 0.5
 
+    ## New in v7.0.0. This item specifies the maximum memory available for the HashAggregation operator with group by key before a disk spill is triggered. When the memory usage exceeds the threshold, HashAggregation reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited and spill to disk is never used for HashAggregation.
+    max_bytes_before_external_group_by = 0
+
+    ## New in v7.0.0. This item specifies the maximum memory available for the sort or topN operator before a disk spill is triggered. When the memory usage exceeds the threshold, the sort or topN operator reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited and spill to disk is never used for topN.
```
Comment (windtalker) [inline tiflash/tiflash-configuration.md:218]:
```suggestion
    ## New in v7.0.0. This item specifies the maximum memory available for the sort or topN operator before a disk spill is triggered. When the memory usage exceeds the threshold, the sort or topN operator reduces memory usage by spilling to disk. This item defaults to 0, which means that the memory usage is unlimited and spill to disk is never used for sort or topN.
```

**Review by windtalker — APPROVED** [review]
Others LGTM

**Review by ran-huang — APPROVED** [review]

**Pull request merged**